src/juju/watchers.ts

```diff
@@ -37,6 +43,23 @@ function generateModelWatcherBase(): ModelData {
   };
 }
 
+function _processDelta(
+  actionType: string,
+  delta: DeltaMessageData,
+  modelData: ModelData,
+  entityType: DeltaEntityTypes,
+  key: string
+): void {
+  if (actionType === "change") {
```

Comment:
"change", "remove" etc. could also be an enum that you use here and when defining the `AllWatcherDelta` tuples.